src/utils/getExtensionId.ts

```diff
@@ -1,20 +1,43 @@
 import * as fs from "fs";
 import * as path from "path";
-import * as vscode from "vscode";
+import { ExtensionContext } from "vscode";
+
+let firstTime = true;
+let extensionId: string | undefined;
+let extensionName: string | undefined;
 
 /**
  * Load the extension id of the package json file
  * @param context Context of VS Code extension
  */
-export const getExtensionId = (
-  context: vscode.ExtensionContext
-): string | undefined => {
-  const extensionPath = path.join(context.extensionPath, "package.json");
+export function getExtensionId(context: ExtensionContext): string | undefined {
+  readPackageFile(context.extensionPath);
+  return extensionId;
+}
+
+/**
+ * Load the extension display name of the package json file
+ * @param context Context of VS Code extension
+ */
+export function getExtensionName(
+  context: ExtensionContext
+): string | undefined {
+  readPackageFile(context.extensionPath);
+  return extensionName;
+}
+
+function readPackageFile(contextExtensionPath: string): void {
+  if (!firstTime) {
+    return;
+  }
+  firstTime = false;
+  const extensionPath = path.join(contextExtensionPath, "package.json");
   const packageFile = JSON.parse(fs.readFileSync(extensionPath, "utf8"));
 
   if (!packageFile) {
-    console.error("Failed to read the extension name from package.json");
-    return undefined;
+    console.error("Failed to read the package.json");
```

Comment:
Wollen wir an dieser Stelle auch den Logger aufrufen? 🧐

Reply:
Geht nicht 😜
Wird verwendet in der Logger inizialisierung. Ich könnte einen Buffer schreiben, der Log messages hällt bis der Logger inizialisiert ist....
oder wir lassen einfach diese beiden Zeilen als `console.error` 😁

Reply:
Alles klar, das passt so :)